HabitTracker/InputValidator.cs

```diff
@@ -0,0 +1,153 @@
+        string input = Console.ReadLine();
+        while (!Int32.TryParse(input, out int parsed))
+        {
+            Console.Write("This is not a valid quantity, please enter a number: ");
+            input = Console.ReadLine();
+        }
+        return Int32.Parse(input);
+    }
+
+    public static int GetHabitId()
+    {
+        string input = string.Empty;
+        DAL dal = new DAL();
+
+        try
+        {
+            string listOfHabits = string.Empty;
+            List<Habit> validHabits = dal.GetHabits();
+            foreach (Habit habit in validHabits)
+            {
+                listOfHabits += $"{habit}\n";
+            }
+            Console.WriteLine(listOfHabits);
+            List<int> validHabitIds = validHabits.Select(h => h.id).ToList();
+            Console.Write("Enter the number corresponding to the habit this entry is for: ");
+            while (true)
+            {
+                if (Int32.TryParse(Console.ReadLine(), out int result))
+                {
+                    if (validHabitIds.Contains(result))
+                    {
+                        return result;
+                    }
+                }
+                Console.Write("This is not a valid id, please enter a number: ");
+            }
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine(e.Message);
+        }
+
+
+        return Int32.Parse(input);
+    }
+
+    public static string GetHabitMeasurement()
+    {
+        Console.Write("What is the measurement of your habit? ");
+        string input = Console.ReadLine();
+        while (input.Length == 0)
+        {
+            Console.Write("Habit measurement should not be empty. Try again: ");
+            input = Console.ReadLine();
+        }
+        return input;
+    }
+
+    public static string GetHabitName()
+    {
+        Console.Write("What is the name of your habit? ");
+        string input = Console.ReadLine();
+        while (input.Length == 0)
+        {
+            Console.Write("Habit name should not be empty. Try a new name: ");
+            input = Console.ReadLine();
+        }
+        return input;
+    }
+
+    public static int GetIdForRemoval()
+    {
+        Console.Write("Which entry do you want to remove? ");
+        DAL dal = new DAL();
+
+        List<int> validIds = dal.GetEntries().Select(o => o.id).ToList();
+        while (true)
+        {
+            if (Int32.TryParse(Console.ReadLine(), out int result))
+            {
+                if (validIds.Contains(result))
+                {
+                    return result;
+                }
+            }
+            Console.Write("This is not a valid id, please enter a number: ");
+        }
+
+    }
+
+    public static int GetIdForUpdate()
+    {
+        Console.Write("Which entry do you want to update? ");
+        DAL dal = new DAL();
+        List<int> validIds = dal.GetEntries().Select(o => o.id).ToList();
+        while (true)
```

Comment:
This applies to anywhere you have the 'while (true)' loop. 
It's not really good practice to have loops like this. Is something happened it could cause an infinite loop and crash the computer. 

But it also doesn't give the user choice. What if the user decided they actually don't want to update this, it would be good for them to have an option to escape back to the main menu or something.

Or at the very least implement something that will make a max attempts. If the user doesn't input a valid input after 5 tries or something, then it will exit.